### PR TITLE
Optimize performance when Git is used as storage repository

### LINKED ISSUE #1101: Consider taking advantage of git commit-graph
With `git commit-graph`, the query `git log Musi` on a repo like [tosdr-snapshots](https://github.com/tosdr/tosdr-snapshots) becomes 42 faster:
```
crawler@ota-tosdr-ubuntu-20-04:~$ git clone https://github.com/tosdr/tosdr-snapshots
Cloning into 'tosdr-snapshots'...
remote: Enumerating objects: 325901, done.
remote: Counting objects: 100% (1/1), done.
remote: Total 325901 (delta 0), reused 0 (delta 0), pack-reused 325900 (from 1)
Receiving objects: 100% (325901/325901), 1014.11 MiB | 23.45 MiB/s, done.
Resolving deltas: 100% (177325/177325), done.
Updating files: 100% (15036/15036), done.
crawler@ota-tosdr-ubuntu-20-04:~$ cd tosdr-snapshots/
crawler@ota-tosdr-ubuntu-20-04:~/tosdr-snapshots$ git --version
git version 2.46.0
crawler@ota-tosdr-ubuntu-20-04:~/tosdr-snapshots$ time git log Musi > /dev/null

real	0m36.150s
user	0m35.476s
sys	0m0.542s
crawler@ota-tosdr-ubuntu-20-04:~/tosdr-snapshots$ time git log Musi/* > /dev/null

real	0m37.725s
user	0m37.163s
sys	0m0.557s
crawler@ota-tosdr-ubuntu-20-04:~/tosdr-snapshots$ git commit-graph write --reachable --changed-paths
Computing commit changed paths Bloom filters: 100% (82187/82187), done.
crawler@ota-tosdr-ubuntu-20-04:~/tosdr-snapshots$ time git log Musi > /dev/null

real	0m0.847s
user	0m0.773s
sys	0m0.073s
crawler@ota-tosdr-ubuntu-20-04:~/tosdr-snapshots$ time git log Musi/* > /dev/null

real	0m38.458s
user	0m38.002s
sys	0m0.453s
crawler@ota-tosdr-ubuntu-20-04:~/tosdr-snapshots$ 
```

But as you can also see there, I haven't found a way yet to speed up the query `git log Musi/*` yet, so unless there is a way to speed that up too, it might be worth to try to avoid using wildcards in paths in git log commands? Will update here if I find out more.

### PR DESCRIPTION
**Pull request by Ndpnt**
This PR implements several optimizations to improve performance and memory usage:

### Technical improvements:

- Replace wildcard patterns in `git log` commands with specific paths
- Enable `commit-graph` for faster commit history traversal (fixes #1101)
- Release HTML/PDF content from memory after term extraction to reduce memory footprint
- Remove unnecessary commit hooks

### Performance benchmarks:

Testing was performed on two distinct datasets:

#### ToS;DR Collection
- Dataset: 100 randomly selected services
- Source: [ToS;DR declarations](https://github.com/tosdr/tosdr-declarations) with [full history](https://github.com/tosdr/tosdr-snapshots)
- Scale: ~92,000 commits, 18,083 terms from 9,202 services

| Operation | Before | After | Improvement |
|-----------|--------|-------|-------------|
| Extraction | 3h 17m 42s | 42s | 99.6% faster |
| Tracking | 1h 19m 8s | 4m 52s | 93.8% faster |

#### P2B Compliance Collection
- Dataset: Complete collection
- Source: [P2B-compliance declarations](https://github.com/OpenTermsArchive/p2b-compliance-declarations) with [full history](https://github.com/OpenTermsArchive/p2b-compliance-snapshots)
- Scale: ~385,000 commits, 245 terms from 203 services

| Operation | Before | After | Improvement |
|-----------|--------|-------|-------------|
| Extraction | 10m 50s | 1m 25s | 87.0% faster |
| Tracking | 16m 15s | 11m 15s | 30.8% faster |

The optimizations show greater impact on collections with larger numbers of declared services and terms.

Fixes #1101. Special thanks to @michielbdejong for reporting the issue, investigating, and providing valuable suggestions 🙏

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 All changes that impact users of this module are documented in this file, in the [Common Changelog](https://common-changelog.org) format with some additional specifications defined in the CONTRIBUTING file. This codebase adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased [patch]
+
+> Development of this release was supported by the [French Ministry for Foreign Affairs](https://www.diplomatie.gouv.fr/fr/politique-etrangere-de-la-france/diplomatie-numerique/) through its ministerial [State Startups incubator](https://beta.gouv.fr/startups/open-terms-archive.html) under the aegis of the Ambassador for Digital Affairs.
+
+### Changed
+
+- Optimize performance when Git is used as storage repository
```
Comment (MattiSG) [inline CHANGELOG.md:11]:
```suggestion
- Optimize performance for Git storage
```

File: src/archivist/index.js [hunk 2]
```diff
@@ -140,7 +144,13 @@ export default class Archivist extends events.EventEmitter {
       return;
     }
 
-    return this.recordVersion(terms, extractOnly);
+    await this.recordVersion(terms, extractOnly);
+
+    terms.sourceDocuments.forEach(sourceDocument => {
```
Comment (MattiSG) [inline src/archivist/index.js:149]: > Why can't we just do `terms.sourceDocuments = []` or `terms.sourceDocuments = null`?

**Review by MattiSG — APPROVED** [review]
Thank you for the benchmarks, this looks amazing!
Thanks a lot @michielbdejong for the experiments and suggestions, I'm thrilled to see this land and am looking forward to the impact this will have on tracking performance across collections!

File: src/archivist/index.js (hunk 2 above)
Comment (Ndpnt) [reply src/archivist/index.js:149]: Because doing so would result in losing information needed for the next run, such as `location`, `contentSelectors`, …

**Pull request merged**